### ISSUE TITLE
The action assumes github.com 

### ISSUE THREAD

**Issue by sadovnikov** [actions/go-dependency-submission#53]
We use GitHub Enterprise Server and therefore GH uses a different hostname.

Seems the action does not use locations from the `github.context` but assumes github.com.

```
2023-05-03T12:40:34.4208205Z ##[error]HTTP Status 401 for request POST https://api.github.com/repos/platform/vs86957-go-test/dependency-graph/snapshots
2023-05-03T12:40:34.4209980Z ##[error]Response body:
{
  "message": "Bad credentials",
  "documentation_url": "https://docs.github.com/rest"
}
2023-05-03T12:40:34.4210878Z ##[error]Bad credentials
2023-05-03T12:40:34.4729810Z ##[error]HttpError: Bad credentials
    at /runner/_work/_actions/actions/go-dependency-submission/v1/webpack:/go-dependency-submission/node_modules/@github/dependency-submission-toolkit/dist/index.js:4235:1
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

**Comment by juxtin** [comment]
Looks like this is better solved in the [dependency-submission-toolkit](https://github.com/github/dependency-submission-toolkit), specifically by adding `baseUrl` [here](https://github.com/github/dependency-submission-toolkit/blob/main/src/snapshot.ts#L187). See the [octokit docs](https://github.com/octokit/request.js#request) for the options reference.

**Comment by lseppala** [comment]
We will update dependency-submission-toolkit to fix this issue and then update this Action. We will close this issue when we confirm the fix. Thank you for reporting!

**Comment by hmaurer** [comment]
This was resolved by https://github.com/github/dependency-submission-toolkit/pull/32, shipped in `dependency-submission-toolkit >= v1.2.5`. We updated `dependency-submission-tookit` to `v1.2.10` in https://github.com/actions/go-dependency-submission/pull/47 but did not release a new version of the action. I released `v1.0.3` just now: https://github.com/actions/go-dependency-submission/releases/tag/v1.0.3. Upgrading to `v1.0.3` should resolve the issue.
